### PR TITLE
Added name field in service account

### PR DESCRIPTION
**Pull request by rohitraut3366**
Added service account name field in `values.yaml`

### CODE REVIEW

**Comment by mattray** [comment]
Did you mean to patch https://github.com/opencost/opencost-helm-chart/blob/main/charts/opencost/templates/serviceaccount.yaml#L5? You'll need to bump the version again, it's currently 1.17.1

**Comment by rohitraut3366** [comment]
This change will make things clear at the values.yaml file level. so users don't have to go in and check how the service account name is being generated.

File: charts/opencost/values.yaml [hunk 1]
```diff
@@ -9,7 +9,9 @@ serviceAccount:
   # --  Annotations to add to the service account
   annotations: {}
   # eks.amazonaws.com/role-arn: arn:aws:iam::123456789012:role/eksctl-opencost
-
+  # The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name: ""
```
Comment (mattray) [inline charts/opencost/values.yaml:14]: I'm not sure serviceAccount.name would get applied, could you verify with
https://github.com/opencost/opencost-helm-chart/blob/main/charts/opencost/templates/serviceaccount.yaml#L5

Comment (rohitraut3366) [reply charts/opencost/values.yaml:14]: Yes, the service name is getting applied correctly with the name field. here is the ref to code https://github.com/rohitraut3366/opencost-helm-chart/blob/rohit/added-name-field-in-service-account/charts/opencost/templates/_helpers.tpl#L72

Comment (dnskr) [reply charts/opencost/values.yaml:14]: The following command tests that `.Values.serviceAccount.name` get applied correctly:
```
helm template my-test charts/opencost --set serviceAccount.name=custom-name
```
Output
```
---
# Source: opencost/templates/serviceaccount.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: custom-name
  labels:
    ...
automountServiceAccountToken: true
---
# Source: opencost/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: my-test-opencost
  labels:
    ...
spec:
  ...
  template:
    metadata:
      labels:
        ...
    spec:
      serviceAccountName: custom-name
      containers:
```

**Review by mattray — APPROVED** [review]

**Pull request merged**